### PR TITLE
flaky_test: Catch SlackApiErrors when attempting to post to the test-flaky channel.

### PR DESCRIPTION
**Pull request by antoniovicente**
Commit Message:
flaky_test: Catch SlackApiErrors when attempting to post to the test-flaky channel.

CI shouldn't fail if posting to this channel fails due to invalid_auth.

Signed-off-by: Antonio Vicente <avd@google.com>

### CODE REVIEW

**Comment by antoniovicente** [comment]
I could use your advice on how to test these changes prior to merge.

File: ci/flaky_test/process_xml.py [hunk 1]
```diff
@@ -273,8 +274,12 @@ def get_git_info(CI_TARGET):
             ssl_context.verify_mode = ssl.CERT_NONE
             # Due to a weird interaction between `websocket-client` and Slack client
             # we need to set the ssl context. See `slackapi/python-slack-sdk/issues/334`
-            client = slack.WebClient(token=SLACKTOKEN, ssl=ssl_context)
-            client.chat_postMessage(channel='test-flaky', text=output_msg, as_user="true")
+            try:
+                client = slack.WebClient(token=SLACKTOKEN, ssl=ssl_context)
+                client.chat_postMessage(channel='test-flaky', text=output_msg, as_user="true")
+            except SlackApiError as e:
```
Comment (phlax) [inline ci/flaky_test/process_xml.py:280]: do we want to swallow this error - ie not raise at all if it fails ?

**Comment by phlax** [comment]
> I could use your advice on how to test these changes prior to merge.

i have been moving python scripts to bazel and added a test framework that works with that - ie avoiding using venv as this script does

afaict this script doesnt have any tests currently, so i wouldnt worry about testing the changes you have made

File: ci/flaky_test/process_xml.py (hunk 1 above)
Comment (phlax) [reply ci/flaky_test/process_xml.py:280]: i guess the answer is in the title

> CI shouldn't fail if posting to this channel fails due to invalid_auth.

**Review by phlax — APPROVED** [review]
not sure on the context to this or if its necessary - but i guess if its causing tests to fail and we are happy to swallow these errors  - then lgtm

**Comment by antoniovicente** [comment]
> not sure on the context to this or if its necessary - but i guess if its causing tests to fail and we are happy to swallow these errors - then lgtm

The context is private repos that have their own CI but no slack credentials.  Failing tests in those repos results in attempts to post to slack which fail and result in CI showing a failure instead of success when there are any flaky tests.

**Comment by phlax** [comment]
> Failing tests in those repos results in attempts to post to slack

should we not avoid this altogether? 

im wondering if we can only post to slack if creds are available

**Comment by antoniovicente** [comment]
> > Failing tests in those repos results in attempts to post to slack
> 
> should we not avoid this altogether?
> 
> im wondering if we can only post to slack if creds are available

It's hard for me to tell what the environment variable is set to in the cases where this was failing.  I wonder if checking for the envionment variable being set but to an empty value would help.  You or @lizan may be more able to tell to what value SLACK_TOKEN is set in private repos.

**Comment by phlax** [comment]
>  what the environment variable is set to in the cases where this was failing. I wonder if checking for the envionment variable being set but to an empty value would help. 

yep - i would just check the variable is non-empty and switch the try...except to a look before you leap pattern

ie, replace...

```
            try:
                client = slack.WebClient(token=SLACKTOKEN, ssl=ssl_context)
                client.chat_postMessage(channel='test-flaky', text=output_msg, as_user="true")
            except SlackApiError as e:
                ...
```

with 

```
             if SLACKTOKEN:
                client = slack.WebClient(token=SLACKTOKEN, ssl=ssl_context)
                client.chat_postMessage(channel='test-flaky', text=output_msg, as_user="true")
```

apart from not attempting to do this in situations where the token is not set - it also prevents this from swallowing errors where it should work

**Comment by phlax** [comment]
hmm - not totally sure if that would work as it seems like `SLACKTOKEN` (and `SLACK_TOKEN` 8/) must be set or it would fail doing `os.environ["SLACK_TOKEN"]` - but i think it will work

```console
$ git grep SLACK_TOKEN
.azure-pipelines/bazel.yml:      SLACK_TOKEN: $(SLACK_TOKEN)
.azure-pipelines/pipelines.yml:              SLACK_TOKEN: $(SLACK_TOKEN)
.azure-pipelines/pipelines.yml:              SLACK_TOKEN: $(SLACK_TOKEN)
ci/flaky_test/process_xml.py:    if os.getenv("SLACK_TOKEN"):
ci/flaky_test/process_xml.py:      SLACKTOKEN = os.environ["SLACK_TOKEN"]
ci/run_envoy_docker.sh:       -e SLACK_TOKEN \

```

afaict `SLACK_TOKEN` is always being set in ci, but possibly to an empty value

**Pull request merged**